### ISSUE TITLE
Is there any way to configure ocLazyLoad with some sort autoload? patterns?

### ISSUE THREAD

**Issue by SchizoDuckie** [ocombe/ocLazyLoad#136]
Hi, I'm planning to convert my project to use ocLazyLoad but i'm a bit turned off by the fact that i'd need to modify a whole lot of files and add injector lookups in many different places. (Or i'm reading the examples wrong)

Is there any way to configure ocLazyLoad in such a way that it scans the dependencies that a module requires, (i'm using ng-strict-di injection style), matches them against a map of configuration data, something like 

``{
  '*Ctrl.js' : 'js/controllers/%s',
  '*Service.js' : function(name) { return name.indexOf('foo') ? 'js/controllers/foo/%s.js' : 'js/controllers/%s.js' },
  'angular-*.js' : 'js/vendor/angular-%s.js'
}`` 

and have it then then automatically resolve and inject them?

I hope this makes more sense than the 140 chars on twitter :)

**Comment by ocombe** [comment]
Hmm there is no built in functionality for that, but it shouldn't be too hard to code a small service that would do that :)
Better yet, what would you say about a node module that would generate a map of your files and that would be able to auto load based on that ? Now **that** would be cool

**Comment by SchizoDuckie** [comment]
That would be fricking sweet! If you can find the time to whip up a prototype for a service that does this (i'm not very familar with hooking angular on that level) i'd be glad to take a stab at creating the node script :)

**Comment by SchizoDuckie** [comment]
I just realize that it could not just do services, but also parse directives and preload their templateUrls (in many cases)

**Comment by ocombe** [comment]
Anything is possible yes, but it would probably make more sense to make a module for webpack

**Comment by SchizoDuckie** [comment]
Not using webpack here :)

(I run from local filesystem so I plan to remove the minify step from my deployment process)

**Comment by lookfirst** [comment]
I'm not really sure I understand the question, but is something like this what you're asking for? https://github.com/lookfirst/systemjs-seed/blob/master/src/common/utils/routing.js

The idea is that it loads code using System.import() based on calls through ui-router.

**Comment by ocombe** [comment]
Yes, it would be similar, but I'd like something that is independent from ui router futurestates

**Comment by ocombe** [comment]
This is on the roadmap for 2.0 https://github.com/ocombe/ocLazyLoad/issues/141
